### PR TITLE
Upgrade to Go 1.21

### LINKED ISSUE #98: Go 1.21 upgrade check list
Once we drop support for 1.20, take a look at nspcc-dev/neo-go#3089.

### PR DESCRIPTION
**Pull request by smallhive**
Closes #98

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,9 @@ This document outlines major changes between releases.
 ### Updated
 - github.com/nspcc-dev/neo-go dependency from v0.106.0 to v0.106.3 (#98)
 - github.com/testcontainers/testcontainers-go dependency from v0.25.0 to v0.32.0 (#98)
+- github.com/oapi-codegen/echo-middleware dependency from v1.0.1 to v1.0.2 (#98)
+- github.com/labstack/echo/v4 dependency from v4.11.4 to v4.12.0 (#98)
+- github.com/getkin/kin-openapi from v0.118.0 to v0.127.0 (#98)
```
Comment (tatiana-nspcc) [inline CHANGELOG.md:15]: I'm not sure that all these package updates should be mentioned in the CHANGELOG. I'd rather add something like "Upgraded Go version to a minimum of 1.21" in the "Changed" section and leave it at that.
But I don't insist.

**Review by roman-khimov — COMMENTED** [review]
This changes the policy for supported versions, please mention that in commits.

File: CHANGELOG.md (hunk 1 above)
Comment (roman-khimov) [reply CHANGELOG.md:15]: These can affect gateway functionality (something like testcontainers can't, so those can be omitted, but it's not a hard requirement to me).

**Comment by smallhive** [comment]
Added lint to new policy

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**